imgparse/imgparse.py

```diff
@@ -708,11 +708,39 @@ def get_ils(image_path, exif_data=None, xmp_data=None):
     try:
         make, model = get_make_and_model(image_path, exif_data)
         xmp_tags = xmp.get_tags(make)
-        return parse_seq(xmp_data[xmp_tags.ILS], float)
+        if make == "DJI":
+            return [float(xmp_data[xmp_tags.ILS])]
+        else:
+            return parse_seq(xmp_data[xmp_tags.ILS], float)
     except KeyError:
         raise ParsingError("Couldn't parse ILS value. Sensor might not be supported")
 
 
+@get_if_needed("exif_data", getter=get_exif_data, getter_args=["image_path"])
+@get_if_needed("xmp_data", getter=get_xmp_data, getter_args=["image_path"])
+def get_irradiance(image_path, exif_data=None, xmp_data=None):
```

Comment:
The M3M, at least the one that I have data for has a DLS which also includes irradiance values, I believe this is corrected using the sun sensor yaw, pitch, roll, but I'm not sure and there isn't a lot of information on it that I've found.